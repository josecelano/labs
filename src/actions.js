/**
 * Add three actions here:
 *
 * - checkItem
 * - addItem
 * - editInput
 */

export function checkItem(text) {
    return {type: 'CHECK_ITEM', text: text}
}

export function addItem(text) {
    return {type: 'ADD_ITEM', text: text}
}

export function editInput(inputValue) {
    return {type: 'EDIT_INPUT', inputValue: inputValue}
}