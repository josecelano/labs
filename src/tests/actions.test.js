/**
 * Tests for the actions
 */

import {checkItem, addItem, editInput } from '../actions';

describe('actions', () => {
    describe('checkItem', () => {
        it('should return the correct constant', () => {
            expect(checkItem('text')).toEqual({type: 'CHECK_ITEM', text: 'text'})
        })
    })

    describe('addItem', () => {
        it('should return the correct constant', () => {
            expect(addItem('text')).toEqual({type: 'ADD_ITEM'})
        })
    })

    describe('editInput', () => {
        it('should return the correct constant', () => {
            expect(editInput('text')).toEqual({type: 'EDIT_INPUT', text: 'text'})
        })
    })
})