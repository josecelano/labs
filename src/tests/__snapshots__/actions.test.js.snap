exports[`addItem should have a type of "ADD_ITEM" 1`] = `
Object {
  "text": undefined,
  "type": "ADD_ITEM",
}
`;

exports[`checkItem should have a type of "CHECK_ITEM" 1`] = `
Object {
  "text": undefined,
  "type": "CHECK_ITEM",
}
`;

exports[`checkItem should pass on the text passed to it as the first argument 1`] = `
Object {
  "text": "Some text",
  "type": "CHECK_ITEM",
}
`;

exports[`editInput should have a type of "EDIT_INPUT" 1`] = `
Object {
  "inputValue": undefined,
  "type": "EDIT_INPUT",
}
`;

exports[`editInput should pass on the text passed to it as the first argument 1`] = `
Object {
  "inputValue": "Some text",
  "type": "EDIT_INPUT",
}
`;
